### ISSUE TITLE
Use prefix to distinguish 'g.' (grafiosch library) from 'gt.' (grafioschtrader application) settings

### ISSUE THREAD

**Issue by hugograf** [grafioschtrader/grafioschtrader#75]
**Which part and what Version?**
Version: 0.36.x
[x] Internal structure (user should not notice)
[ ] Improvement for the user

**Problem**

The original goal of this issue — removing the `gt.datafeed` prefix from `id_connector` — is **obsolete and dropped**. Removing the prefix is no longer desirable, and `gt.datafeed.` is retained (see below).

The project has grown into two clearly separated layers:

- **grafiosch** — a generic, reusable library (`grafiosch-base`, `grafiosch-server-base`)
- **grafioschtrader** — the concrete application built on top of it (`grafioschtrader-common`, `grafioschtrader-server`, Angular `frontend`)

The configuration prefix should express this layering so that a downstream application can override generic library defaults while keeping its own application-specific keys clearly separated:

- **`g.`** = generic defaults owned by the grafiosch **library**
- **`gt.`** = settings specific to the grafioschtrader **application**

This convention is only **partially implemented**. The scaffolding already exists:

- `grafiosch-base` → `BaseConstants.G_PREFIX = "g."`; `BaseConstants.PREFIXES_PARAM` is seeded with `g.` and documented as *"A dependent application should extend this prefix accordingly."*
- `grafioschtrader-common` → `GlobalConstants.GT_PREFIX = "gt."`
- `BaseFeatureConfig` uses `@ConfigurationProperties(prefix = "g.use")` and `FeatureConfig` uses `@ConfigurationProperties(prefix = "gt.use")` — both correct.
- Most library keys are already correct: `g.jwt.expiration.minutes`, `g.alert.mail.bitmap`, `g.gnet.*`, `g.limit.day.*`, `g.task.data.days.preserve`, `g.purge.mail.role.cron`, `g.gnet.offline.announce`.

However, a number of **library-owned** keys still carry the `gt.` prefix. This is inconsistent, blocks clean reuse of the grafiosch library by other applications, and blurs the line between generic defaults and application settings.

**Solution / Implementation**

Adopt and fully enforce the following rule:

> A configuration key, NLS message key, `globalparameters` property name, or `@Value` / `@ConfigurationProperties` property is prefixed by the **module layer that owns it**:
> - Defined in or read by `grafiosch-base` / `grafiosch-server-base` → **`g.`**
> - Specific to `grafioschtrader-common` / `grafioschtrader-server` / `frontend` → **`gt.`**
>
> The library provides `g.`-prefixed defaults; the application may override them and adds its own `gt.` keys.

---

### Already done (was previously listed here as open)

- `GlobalParamKeyBaseDefault.GNET = BaseConstants.G_PREFIX + "gnet."`, so `GLOB_KEY_GTNET_DEL_MESSAGE_RECV` and `GLOB_KEY_GTNET_LOG_AGGREGATE_DAYS` already resolve to `g.gnet.del.message.recv` / `g.gnet.log.aggregate.days` (`GlobalParamKeyBaseDefault.java:80,99,103`).
- `V0_36_2__import_rounding_tax_correction_minor_unit_gnet_prefix.sql:40-41` migrates both `globalparameters` rows.
- `GTStartUp.java:105` extends `BaseConstants.PREFIXES_PARAM` with `GlobalConstants.GT_PREFIX`.
- The deployment migration script and its integration into the update flow are done, see *Deployment migration* below.

### 1. Deployment properties (`@Value` / `@Scheduled`) — the breaking part

The list previously in this issue was incomplete; it named five keys, there are nine. All of them are consumed by `grafiosch-server-base`. `grafiosch-base` contains no configuration placeholder at all.

| Legacy property | New property | Library consumer |
| --- | --- | --- |
| `gt.jwt.secret` | `g.jwt.secret` | `JwtConfig:33` |
| `gt.main.user.admin.mail` | `g.main.user.admin.mail` | `UserServiceImpl:73`, `AlertListener:51`, `SendMailInternalExternalService:91` |
| `gt.allowed.users` | `g.allowed.users` | `UserServiceImpl:76`, `UserJpaRepositoryImpl:46` |
| `gt.demo.account.pattern.de` | `g.demo.account.pattern.de` | `UserServiceImpl:79`, `TenantBaseImpl:45` |
| `gt.demo.account.pattern.en` | `g.demo.account.pattern.en` | `UserServiceImpl:82`, `TenantBaseImpl:48` |
| `gt.purge.cron.expression` | `g.purge.cron.expression` | `TokensPurgeTask:30` |
| `gt.purge.task.data` | `g.purge.task.data` | `TaskDataChangePurgeTask:24` |
| `gt.gtnet.log.aggregation.cron` | `g.gnet.log.aggregation.cron` | `GTNetExchangeLogAggregationAndDelGTNetMessagesTask:73` |
| `gt.gtnet.future.message.cron` | `g.gnet.future.message.cron` | `GTNetFutureMessageDeliveryTask:125` |

**Out of scope, stays `gt.`:** `gt.datafeed.*` (see below), `gt.gtnet.exchange.sync.cron`, `gt.eod.*`, `gt.dividend.*`, `gt.calendar.*`, `gt.use.*`, `gt.connector.*` and the other connector settings — all consumed by `grafioschtrader-server`.

Files that must move in lockstep with the Java change:

- `grafiosch-server-base/src/main/resources/grafiosch-application.properties:15,17` — the **library's own defaults file**, easily overlooked; it defines `gt.purge.cron.expression` and `gt.purge.task.data`.
- `grafioschtrader-server/src/main/resources/application.properties:76,96,105,106,127,130,152`
- `grafiosch-test-integration/src/main/resources/application.properties:11-16`
- `grafioschtrader-server/src/main/resources/META-INF/additional-spring-configuration-metadata.json:9,21` — declares the group `gt.jwt` and the property `gt.jwt.secret`; while renaming, also fix its `sourceType`, which still names the no-longer-existing `grafioschtrader.security.JwtTokenHandler` instead of `grafiosch.security.JwtConfig`.
- `docker/docker-compose.yml:36-38`, `docker/backend.Dockerfile:29`
- `backend/README.md:21-23`, `CLAUDE.md:132,160`

Note that `gt.gtnet.log.aggregation.cron` is defined in **no** properties file — it exists only as the default of its `@Scheduled` annotation.

### 2. Library NLS message keys still using `gt.`

The list below is by example only. The implementation must sweep **every** `gt.` literal in both generic modules, not just these:

- `grafiosch-base/.../dto/InputRule.java` → `gt.input.rule.{min,max,enum,pattern}.violation` → `g.input.rule.*`
- GTNet validation keys in `grafiosch-base`: `GTNetJpaRepositoryImpl.java`, `GTNetExchangeLogJpaRepositoryImpl.java` (`gt.gtnet.log.period.*`)
- `TaskDataChangeJpaRepositoryImpl.java` → `gt.taskdatachange.user.definition`
- `gt.afterEqual`, `gt.external.message.addition`, `gt.password.regex`, `gt.tenant.access.readonly`, `gt.move.createdby.procedure.failure`, `gt.demo.func.not.available`, `gt.evalex.*` and further `gt.gtnet.*` message keys
- The matching entries in the four message bundles (UTF-8 without BOM — keep the German umlauts intact), plus `nls-mapping.properties` and any test that asserts on a key.

Where a message is consumed by `grafiosch-server-base` but currently lives in the Grafioschtrader bundle, move it into the library bundle at the same time.

### 3. globalparameters DB rows

Any `property_name` matching a renamed library key must be migrated via a **Flyway migration**:

```sql
UPDATE globalparameters SET property_name = 'g....' WHERE property_name = 'gt....';
```

Migration version follows the current pom version + highest existing migration in that series (append to an unreleased migration if one is open on the branch). Keep it idempotent.

### 4. Deployment migration — already shipped, no action needed at rename time

Renaming a deployment property silently destroys user configuration on update:

- `merger.sh` matches keys by exact string equality. A user file holding `gt.jwt.secret=ENC(...)` against a template holding `g.jwt.secret` finds no match, so the **template placeholder is written and the user's encrypted secret is lost**.
- `gtupdate.sh` does not merge `application-production.properties` at all — it copies the backup back verbatim, so a renamed key there is silently ignored and the built-in default takes over.

`util/shellscripts/gt_to_g_rename.sh` handles both. It maps exactly the nine keys above, preserves values, indentation, spacing and the `#`/`!` comment state, is idempotent, and refuses to guess when a file contains both spellings of a key.

It is **self-detecting**: it reads the `application.properties` of the freshly checked-out release and applies the mapping only once that template actually uses the new `g.*` names. It is therefore already wired into `gtupdate.sh` and `docker/update.sh` and is a documented no-op until this issue is implemented — at which point it switches itself on with no further change to the update scripts.

Consequence for whoever implements the rename: nothing has to be done to the update flow. Just make sure the shipped `application.properties` carries the new key names, which is part of section 1 anyway.

### 5. Still required in the rename release: backend compatibility bridge

Docker installations come from a shallow clone and `docker/update.sh` only pulls new **images** — it never refreshes `docker-compose.yml` or itself. An installed host therefore keeps supplying `GT_JWT_SECRET`, `GT_MAIN_USER_ADMIN_MAIL` and `GT_ALLOWED_USERS` while running a new image that reads only `g.*`. Depending on the built-in defaults the backend then fails to start or silently uses the wrong administrator and user limit.

The first release with the rename must therefore accept the legacy names:

- An `EnvironmentPostProcessor` in `grafiosch-server-base`, registered via `META-INF/spring.factories`, logging through `DeferredLogFactory`.
- Walk `MutablePropertySources` in priority order. For each of the nine mappings, find the first source defining *either* name. If that source defines only the legacy name, add the value under the new name with `addFirst` — safe precisely because no higher-priority source defined the new name.
- If one source defines both, the new `g.*` name wins.
- Log a deprecation warning when a legacy name is used.
- Keep for at least one documented upgrade release.

A nested `@Value("${g.jwt.secret:${gt.jwt.secret}}")` does **not** work: the packaged `application.properties` defines the new key, so the outer placeholder is always satisfied and the fallback never fires. For the same reason the bridge must inspect individual property sources rather than calling `Environment.getProperty(newName)`.

**Compose rename:** only the left-hand side changes (`G_JWT_SECRET`, `G_MAIN_USER_ADMIN_MAIL`, `G_ALLOWED_USERS`). The user-facing `.env` names `JWT_SECRET`, `ADMIN_EMAIL` and `GT_ALLOWED_USERS` stay, so existing `.env` files need no edit. `GT_ACME_EMAIL` (Caddy) and `GT_USE_WEBSOCKET` are unrelated to Spring binding and stay as they are.

⚠️ Flag the whole rename prominently in the release notes as a breaking configuration change, even with the bridge in place.

### 6. Unrelated defects found while reviewing this issue

- `grafiosch-test-integration/migration-baseline/V0_10_0__init.sql:576,579` still seeds `gt.gtnet.del.message.recv` and `gt.gtnet.log.aggregate.days`, while the code reads `g.gnet.*` and that environment never runs `V0_36_2`. Both rows are dead there and the hardcoded defaults are used instead. Lines 559-562 of the same file already use the new names for the other four GTNet keys.
- `grafiosch-server-base/.../TenantLimitsHelper.java:32,222` reference `BaseConstants.GT_PREFIX`, which does not exist (`GT_PREFIX` lives in `GlobalConstants`) — javadoc and commented-out code.
- `backend/README.md:23` documents `app.jwt.secret`, a name that no longer exists.

### 7. Verification checklist for the rename release

The migration tooling of section 4 is already verified (renaming with values, markers, indentation and spacing preserved; idempotency; conflict abort across several files; the classic `gtupdate.sh` flow and `docker/update.sh` simulated end to end, including the control case proving the JWT secret is lost without it). What still has to be checked when the rename itself is implemented:

- [ ] Start the new backend image with an **old** Compose file (`GT_JWT_SECRET`, `GT_MAIN_USER_ADMIN_MAIL`, `GT_ALLOWED_USERS`) and confirm each legacy variable is accepted, with a deprecation warning in the log.
- [ ] Start it with both old and new variables and confirm the `G_*` value wins.
- [ ] Start a fresh Docker installation with the updated Compose file and confirm the unchanged `.env` keys (`JWT_SECRET`, `ADMIN_EMAIL`, `GT_ALLOWED_USERS`) still supply `g.jwt.secret`, `g.main.user.admin.mail` and `g.allowed.users`.
- [ ] Confirm the classic update on an installed machine carries the encrypted `g.jwt.secret` through `gt_to_g_rename.sh` + `merger.sh` — this is the failure case the migration exists for.
- [ ] Confirm `application-production.properties` is migrated too; `gtupdate.sh` copies it back unmerged, so a key left with the old name there fails silently.
- [ ] Confirm Flyway migrates the renamed `globalparameters` rows and that no `gt.*` library key survives in the database.
- [ ] Grep the built jar and both message bundles for any remaining `gt.` literal owned by the library.

**Explicitly out of scope (kept as-is):** the `gt.datafeed.` connector ID prefix (`BaseFeedConnector.ID_PREFIX`). Connectors live entirely in the application layer (`grafioschtrader-server`), so `gt.datafeed.` is the *correct* application prefix. It is also persisted in `securitycurrency.id_connector_history` / `id_connector_intra` and in `globalparameters` connector defaults, so it must not be changed.

**Follow-up (separate issue):** `docker/update.sh` cannot update `docker-compose.yml` or itself, so no change to the Compose file ever reaches an existing installation. A bundle-update mechanism that preserves locally edited Compose files is needed independently of this issue.

The `g.` / `gt.` prefix convention will be documented in `backend/CLAUDE.md` so it is adhered to for all future keys.

**Comment by hugograf** [comment]
Released in V0.36.9. Source review confirmed the nine library-owned deployment properties now use the g.* prefix, the classic migration helper and guards are present, and current Docker Compose uses the G_* environment wiring. The missing automatic refresh of deployment files on existing Docker hosts—and the live upgrade evidence—is tracked separately in #240.
